### PR TITLE
Integrate lump-sum logic

### LINKED ISSUE #22: Lump-sum to monthly pension calculation

### PR DESCRIPTION
**Pull request by alexjcode**
- Add lump sum actuarial values to wep-cache
- Fix typo

closes #22 #67

### CODE REVIEW

**Comment by alexjcode** [comment]
~The date picker is not hiding when i select a date for `dateAwarded`~
fixed (https://github.com/Hacker0x01/react-datepicker/issues/1012)

**Comment by alexjcode** [comment]
It seems to put out the correct values even when scrolling back and forth through menus and selecting different options.

**Comment by thadk** [comment]
Looks good on first try. 

@supjohn Can you maybe give this a try at the preview and see if the values seem consistent with your logic? https://deploy-preview-111--windfall-develop.netlify.com/ 

Here was the result of one test run Alex posted on Slack:
```
// Sample User
DOB => 12/3/1955
Retirement Age => 65
Earnings => http://ssa.gov/osss/schemas/1.0/samples/Full_Retirement.xml
// Do you have earnings that do not show up on your Social Security record?
No =>
  $1716.97 Monthly benefit at full retirement age
Yes =>
  Monthly Pension @ $15000
    => $1308.89 Monthly benefit at full retirement age
  Lump Sum (12/3/2016) @ $15000
    => $1680.30 Monthly benefit at full retirement age
  Other
    => 1716.97 Monthly benefit at full retirement age
```

I'll give a little code review too. Thanks Alex!

**Comment by alexjcode** [comment]
~Just for completeness, I should mention that the last year in that [sample earnings record](http://ssa.gov/osss/schemas/1.0/samples/Full_Retirement.xml) should be `0` not `-1`~

**Comment by thadk** [comment]
@alexjcode We got that XML file from the SSA, I think -1 must be a coded value for year after last year of earnings?

File: src/library/observable-functions.js [hunk 1]
```diff
@@ -437,6 +452,80 @@ function getFullRetirementDate(dob) {
...
+// Lump Sum to Monthly Pension Converter //
+///////////////////////////////////////////
+
+function lumpSumToMonthly(lumpSum, dateAwarded, workerAge, dob){
+  let ageAwarded = dateAwarded.diff(dob, "year")
+	let actuarialValue = getActuarialValue(dateAwarded, workerAge, ageAwarded);
+	let monthlyPension = Math.floor(lumpSum*100 / actuarialValue.result)/100.00;
+	console.log('actuarialValue', actuarialValue)
+  console.log('monthlyPension', monthlyPension)
+	// return ({...actuarialValue, monthlyPension}); (not sure why it was formatted this way)
```
Comment (thadk) [inline src/library/observable-functions.js]: Please remove all the console.log()'s.

I think I did the multi-part return so there was more debug information available. It is good you were able to remove the extra things.

File: src/library/wep-tables.js [hunk 2]
```diff
@@ -12,6 +12,19 @@ var getWepTables = (function() {
 
 		return ColaTable
 	}
+	
+	var actuarialValueLumpSumTable = function(){
+		const actuarialValueLumpSumTable = checkJSONCache("actuarialValueLumpSumTable") || d3.csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vQ0KGd5_ZHdYIIFPdSzBHkG38FAkGQ04ADGHKkT-ODA64Qr79Ia8g-W1DMNFb4LEC2sh502ybgPeQDF/pub?gid=1500974875&single=true&output=csv',
```
Comment (thadk) [inline src/library/wep-tables.js:17]: Looks great! Thanks for figuring this out.

File: src/pages/prescreen-1c.tsx [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import React from "react";
 import styled from "@emotion/styled";
+import DatePicker from "react-datepicker";
+// import "react-datepicker/dist/react-datepicker.css";
```
Comment (thadk) [inline src/pages/prescreen-1c.tsx]: you can just erase the `// import...` I think, thanks for noticing this bug.

File: src/pages/screen-2.tsx [hunk 4]
```diff
@@ -77,7 +80,50 @@ export default class Screen2 extends React.Component {
       this.state.userWEP = false;
       let userWEP = false;
     }
-    var userPension = Number(SessionStore.get("pensionAmount"))
+    
+    // //////////////////////////////////////
+    
+    // let dob = dayjs(dayjs(SessionStore.get("BirthDate")).format(), dayjs.ISO_8601)
+    // console.log('dob-incoming', userDOB)
+    // console.log('dob', dob)
```
Comment (thadk) [inline src/pages/screen-2.tsx]: Please remove the extra comments before the final version too.

**Comment by thadk** [comment]
Anne shared this case on Apr 23, 2019 (exact copy below) 
https://cfb-public.slack.com/archives/CG08M4KN3/p1556066757027200
```
Example case
DOB: June 20 1948
therefore entitlement starts July 2010
AIME indexing year: 2008
AIME: $2,212.00
first factor in WEP: 40%
She also got three COLA increases: 3.6% in Dec 2011, 1.7% in Dec 2012, and 1.5% in Dec 2013
Lump sum of pension: $22,578.44, eligible from June 30 2014; returns $180.27 monthly pension amount
SSA’s WEP adjusted PIA: $768.70
COLA adjustment: $53.20
Total max payable PIA @ Full retirement age: $821.90
```

Maybe you can run the function manually with that AIME plugged in partway somehow?

File: src/pages/screen-2.tsx [hunk 5]
```diff
@@ -77,7 +80,50 @@ export default class Screen2 extends React.Component {
...
+        userPension = ObsFuncs.lumpSumToMonthly(
+          SessionStore.get("pensionAmount"),
+          dayjs(new Date(JSON.parse(SessionStore.get("dateAwarded"))).toLocaleDateString("en-US")),
+          workerAge,
+          dob
+        )
+        console.log('userPension', userPension)
+      } else {
+        console.log('other')
+        userPension = Number("0")
```
Comment (thadk) [inline src/pages/screen-2.tsx]: We should avoid setting pension to zero when it is really `undefined`. In the [upcoming version](https://github.com/codeforboston/windfall-elimination/pull/130) I have a placeholder in that field that turns red when it is actually `undefined` so users aren't caught off guard when users start typing and to make it clear they have to type something if they say they have pension/account:
image

**Review by thadk — CHANGES_REQUESTED** [review]

File: src/pages/screen-2.tsx (hunk 5 above)
Comment (alexjcode) [reply src/pages/screen-2.tsx]: ~Would it be better to remove or hide the row?~ Seems like this will be covered separately

**Comment by alexjcode** [comment]
Thanks @thadk I will make those changes.

File: src/pages/screen-2.tsx (hunk 5 above)
Comment (alexjcode) [reply src/pages/screen-2.tsx]: I changed it to undefined, and it works

**Review by thadk — APPROVED** [review]

File: src/components/file-upload.jsx [hunk 6]
```diff
@@ -162,13 +162,13 @@ export class GenerateTable extends React.Component {
     }
     const displayFile=this.props.fileName ? "from " + this.props.fileName : ""
 
-    return (
+    return (SessionStore.get("earnings") ? (
```
Comment (thadk) [inline src/components/file-upload.jsx:165]: I had to revert this to fix the bug in manual entry I mentioned in Slack.

**Pull request merged**